spec/unit/matrixrtc/MembershipManager.spec.ts

```diff
@@ -0,0 +1,588 @@
+                await flushPromises();
+                // reset all mocks before checking what happens when calling: `onRTCSessionMemberUpdate`
+                (client.sendStateEvent as Mock).mockClear();
+                (client._unstable_updateDelayedEvent as Mock).mockClear();
+                (client._unstable_sendDelayedStateEvent as Mock).mockClear();
+                waitForMockCall(client._unstable_sendDelayedStateEvent, { delay_id: "id" });
+
+                // our own membership is removed:
+                manager.onRTCSessionMemberUpdate([mockCallMembership(membershipTemplate, room.roomId)]);
+                await flushPromises();
+                expect(client.sendStateEvent).toHaveBeenCalled();
+                expect(client._unstable_sendDelayedStateEvent).toHaveBeenCalled();
+
+                expect(client._unstable_updateDelayedEvent).toHaveBeenCalled();
+            });
+        });
+
+        // TODO: not sure about this name
+        describe("background timers", () => {
+            it("sends only one keep-alive for delayed leave event per `membershipKeepAlivePeriod`", async () => {
+                const manager = new TestMembershipManager(
+                    { membershipKeepAlivePeriod: 10_000, membershipServerSideExpiryTimeout: 30_000 },
+                    room,
+                    client,
+                    () => undefined,
+                );
+                waitForMockCall(client._unstable_sendDelayedStateEvent, {
+                    delay_id: "id",
+                });
+                manager.join([focus], focusActive);
+                expect(client._unstable_sendDelayedStateEvent).toHaveBeenCalledTimes(1);
+
+                // The first call is from checking id the server deleted the delayed event
+                // so it does not need a `advanceTimersByTime`
+                await flushPromises();
+                expect(client._unstable_updateDelayedEvent).toHaveBeenCalledTimes(1);
+                // TODO: check that update delayed event is called with the correct HTTP request timeout
+                // expect(client._unstable_updateDelayedEvent).toHaveBeenLastCalledWith("id", 10_000, { localTimeoutMs: 20_000 });
+
+                for (let i = 2; i <= 12; i++) {
+                    // flush promises before advancing the timers to make sure schdulers are setup
+                    await flushPromises();
+                    jest.advanceTimersByTime(10_000);
+                    expect(client._unstable_updateDelayedEvent).toHaveBeenCalledTimes(i);
+                    // TODO: check that update delayed event is called with the correct HTTP request timeout
+                    // expect(client._unstable_updateDelayedEvent).toHaveBeenLastCalledWith("id", 10_000, { localTimeoutMs: 20_000 });
+                }
+            });
+
+            it("extends `expires` when call still active !FailsForLegacy", async () => {
```

Comment:
Can we add a comment to say why this FailsForLegacy? I think you mentioned that the code was removed at some point? Could we reference the commit?